frame/staking/src/lib.rs

```diff
@@ -688,33 +695,50 @@ decl_storage! {
 		/// Direct storage APIs can still bypass this protection.
 		Nominators get(fn nominators): linked_map T::AccountId => Option<Nominations<T::AccountId>>;
 
-		/// Nominators for a particular account that is in action right now. You can't iterate
-		/// through validators here, but you can find them in the Session module.
-		///
-		/// This is keyed by the stash account.
-		pub Stakers get(fn stakers): map T::AccountId => Exposure<T::AccountId, BalanceOf<T>>;
-
-		/// The currently elected validator set keyed by stash account ID.
-		pub CurrentElected get(fn current_elected): Vec<T::AccountId>;
-
 		/// The current era index.
+		///
+		/// This is the latest planned era, depending on how session module queues the validator
+		/// set, it might be active or not.
 		pub CurrentEra get(fn current_era) config(): EraIndex;
 
-		/// The start of the current era.
-		pub CurrentEraStart get(fn current_era_start): MomentOf<T>;
+		/// The active era index, the era currently rewarded.
+		pub ActiveEra get(fn active_era) config(): EraIndex;
 
-		/// The session index at which the current era started.
-		pub CurrentEraStartSessionIndex get(fn current_era_start_session_index): SessionIndex;
+		/// The start of the currently rewarded era.
+		pub ActiveEraStart get(fn active_era_start): MomentOf<T>;
 
-		/// Rewards for the current era. Using indices of current elected set.
-		CurrentEraPointsEarned get(fn current_era_reward): EraPoints;
+		/// Exposure of validator at era.
+		///
+		/// This is keyed fist by the era index to allow bulk deletion and then the stash account.
+		///
+		/// Is it removed after `HISTORY_DEPTH` eras.
+		pub ErasStakers get(fn eras_stakers):
+			double_map hasher(twox_64_concat) EraIndex, hasher(twox_64_concat) T::AccountId
+			=> Exposure<T::AccountId, BalanceOf<T>>;
+		// TODO: consider switching this to a simple map EraIndex => Vec<Exposure>
```

Comment:
We can do this in a future PR, also I feel like it is hard to benchmark things currently in substrate

Reply:
Note that having this structure as a `Vec<Exposure>` means decoding all validator exposure (which can contains a lot of nominators) when rewarding a single nominator and validator.
So instead of doing at most 16 (MAX_NOMINATION) exposure read we always do 1 read of all exposure.